altapay.php

```diff
@@ -2535,4 +2505,23 @@ public function returnError($paymentID, $exception)
 
         return $cookie->altapayError;
     }
+
+    /**
+     * @param int $itemID
+     * @param int|float $cmsTotal
+     * @param int|float $gatewayTotal
+     *
+     * @return array
+     */
+    public function compensationOrderlines($itemID, $cmsTotal, $gatewayTotal)
+    {
+        $orderLines['description'] = 'compensation'; // Description of item.
+        $orderLines['itemId'] = 'comp-' . $itemID; // Item number (SKU)
+        $orderLines['quantity'] = 1;
+        $orderLines['unitPrice'] = $cmsTotal - $gatewayTotal;
```

Comment:
This is where you should do a `number_format()` if `unitPrice` expects a string.